src/Extensions/LeftAndMainSubsites.php

```diff
@@ -86,49 +85,38 @@ public function sectionSites($includeMainSite = true, $mainSiteTitle = 'Main sit
             $member = DataObject::get_by_id(Member::class, $member);
         }
 
-        // Collect permissions - honour the LeftAndMain::required_permission_codes, current model requires
-        // us to check if the user satisfies ALL permissions. Code partly copied from LeftAndMain::canView.
-        $codes = [];
-        $extraCodes = Config::inst()->get(get_class($this->owner), 'required_permission_codes');
-        if ($extraCodes !== false) {
-            if ($extraCodes) {
-                $codes = array_merge($codes, (array)$extraCodes);
-            } else {
-                $codes[] = sprintf('CMS_ACCESS_%s', get_class($this->owner));
-            }
-        } else {
-            // Check overriden - all subsites accessible.
-            return Subsite::all_sites();
-        }
+        $accessibleSubsites = ArrayList::create();
+        $subsites = Subsite::all_sites($includeMainSite, $mainSiteTitle);
 
-        // Find subsites satisfying all permissions for the Member.
-        $codesPerSite = [];
-        $sitesArray = [];
-        foreach ($codes as $code) {
-            $sites = Subsite::accessible_sites($code, $includeMainSite, $mainSiteTitle, $member);
-            foreach ($sites as $site) {
-                // Build the structure for checking how many codes match.
-                $codesPerSite[$site->ID][$code] = true;
-
-                // Retain Subsite objects for later.
-                $sitesArray[$site->ID] = $site;
-            }
+        // Check whether we have any subsites
+        if (!$subsites->exists()) {
+            return $accessibleSubsites;
```

Comment:
Yeah you're right since it returns an ArrayList which includes a stub for the main site. I think I wrote this when it was a DataList (which wouldn't). I've removed it anyway, it's not needed